### ISSUE TITLE
ext_vector_type ternary causes ISEL assertion when using bool & char.

### ISSUE THREAD

**Issue by erichkeane** [llvm/llvm-project#55038]
See: https://godbolt.org/z/8jbTPd5vv

I suspect the issue is that the bool vector is i1s vs the char being i8, yet the two have the same CFE size.  This might need to be fixed in the CFE.

Indeed you see IR like that here:   

%21 = and <4 x i8> %18, <4 x i1> %20, !dbg !32
  %22 = and <4 x i8> %17, <4 x i1> %19, !dbg !32

**Comment by fhahn** [comment]
Yeah, it looks like clang generates an invalid `AND` instruction for the code below. I updated the title.

```
typedef bool bool4 __attribute__((ext_vector_type(4)));
typedef char char4 __attribute__((ext_vector_type(4)));

void foo(bool4 mask, char4 one, char4 two) {
    auto output = mask ? one : two;
}
```

`clang -emit-llvm -g0`

```
...
  %21 = and <4 x i8> %18, <4 x i1> %20
  %22 = and <4 x i8> %17, <4 x i1> %19
...
```

**Comment by kaz7** [comment]
@simoll, do you notice this error report?  Thanks!

**Comment by c-rhodes** [comment]
reproduced on 15, fixed in 22 and on trunk: https://godbolt.org/z/z7hcGhfz9

Fix was #154145.
